v2/pkg/engine/datasource/graphql_datasource/graphql_datasource.go

```diff
@@ -343,11 +350,34 @@ func (p *Planner[T]) ConfigureFetch() resolve.FetchConfiguration {
 		postProcessing = EntitiesPostProcessingConfiguration
 	}
 
+	var dataSource resolve.DataSource
+
+	dataSource = &Source{httpClient: p.fetchClient}
+
+	if p.config.grpc != nil {
+		var err error
+
+		opDocument, opReport := astparser.ParseGraphqlDocumentBytes(operation)
+		if opReport.HasErrors() {
+			p.stopWithError(errors.WithStack(fmt.Errorf("failed to parse operation: %w", opReport)))
+			return resolve.FetchConfiguration{}
+		}
+
+		dataSource, err = grpcdatasource.NewDataSource(p.grpcClient, grpcdatasource.DataSourceConfig{
+			Operation:  &opDocument,
+			Definition: p.config.schemaConfiguration.upstreamSchemaAst,
+			Mapping:    p.config.grpc.Mapping,
+			Compiler:   p.config.grpc.Compiler,
```

Comment:
do we need subgraph name here?

If yes it could be obtained like this

```go
SubgraphName: p.dataSourceConfig.Name(),
```

Reply:
We should not be relying on it as we have the mapping in place, it's currently rather a fallback if the mapping name is missing.

I think I'll add it in for now but also put a TODO there, that only mapping should be used